x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/metadata.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiLoadingChart } from '@elastic/eui';
+import { EuiText } from '@elastic/eui';
+import { useSourceContext } from '../../../../../../containers/metrics_source';
+import { findInventoryModel } from '../../../../../../../common/inventory_models';
+import type { InventoryItemType } from '../../../../../../../common/inventory_models/types';
+import { useMetadata } from '../../../../metric_detail/hooks/use_metadata';
+import { Table } from './table';
+import { getAllFields } from './utils';
+import type { HostNodeRow } from '../../../hooks/use_hosts_table';
+import type { MetricsTimeInput } from '../../../../metric_detail/hooks/use_metrics_time';
+
+const NODE_TYPE = 'host' as InventoryItemType;
+
+export interface TabProps {
+  currentTimeRange: MetricsTimeInput;
+  node: HostNodeRow;
+}
+
+const Metadata = ({ node, currentTimeRange }: TabProps) => {
+  const nodeId = node.name;
+  const inventoryModel = findInventoryModel(NODE_TYPE);
+  const { sourceId } = useSourceContext();
+  const { loading: metadataLoading, metadata } = useMetadata(
```

Comment:
**thought:** we are not handling the error in case the metadata request fail, which would simply not return items and tell the user that no metadata has been found.
Should we handle this case properly and give feedback to the user about what didn't work?
@kkurstak any idea on what we want to do here?

Reply:
@kkurstak @tonyghiani Should I create an issue and we can add a design for it? If we just want to let's say show a generic callout for "Something went wrong, please check {{error}}" or just "failed to fetch data" I can also add it here 🤔 Usually the metadata should be available but I really like the idea to have this error case handled

Reply:
Just a side note: I checked inventory to get an idea of how this message should look but I don't see it implemented there. So we can also use that error handling in inventory as well.